runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/internal/AwsChunkedUtil.kt

```diff
@@ -43,8 +43,8 @@ internal fun SdkBuffer.writeTrailerSignature(signature: String) {
  */
 @InternalApi
 public val HttpBody.isEligibleForAwsChunkedStreaming: Boolean
-    get() = (this is HttpBody.SourceContent || this is HttpBody.ChannelContent) && contentLength != null &&
-        (isOneShot || contentLength!! > AWS_CHUNKED_THRESHOLD)
+    get() = (this is HttpBody.SourceContent || this is HttpBody.ChannelContent) &&
+        (isOneShot || (contentLength?.compareTo(AWS_CHUNKED_THRESHOLD) ?: 0) > 0)
```

Comment:
**Style**: `?.compareTo` works but is pretty verbose and still requires null-coalescing the value. Shorter would just be:

```kotlin
(contentLength ?: 0 > AWS_CHUNKED_THRESHOLD)
```

Same comment applies below.